File: packages/sdk/js-example/src/batch-async.ts
import type { CompletionBatchRequest } from '@vm-x-ai/sdk';
import { VMXClient } from '@vm-x-ai/sdk';

const main = async () => {
  const client = new VMXClient();
  console.log('Starting batch');

  const requests = Array(100)
    .fill(0)
    .map<CompletionBatchRequest['requests'][number]>(() => ({
      messages: [
        {
          role: 'user',
          content: 'Hi',
        },
      ],
      resource: 'default',
    }));

  const response = await client.completionBatch(requests);

  console.log('Batch created', response.batchId);

  const result = await client.waitForCompletionBatch(response.batchId);
  console.log('Batch completed', result);
};

main()
  .catch((error) => {
    console.error(error);
    process.exit(1);
  })
  .finally(() => process.exit(0));

File: packages/sdk/python-example/example/batch_async.py
#!/usr/bin/env python
import asyncio

from vmxai import VMXClient
from vmxai.types import CompletionRequest, RequestMessage


async def main():
    """Example demonstrating how to submit batch completion requests asynchronously."""
    client = VMXClient()
    print("Starting batch")

    # Create 100 completion requests
    requests = [
        CompletionRequest(
            messages=[
                RequestMessage(
                    role="user",
                    content="Hi",
                )
            ],
            resource="default",
        )
        for _ in range(10)
    ]

    # Submit the batch asynchronously
    response = await client.completion_batch(requests)
    print(f"Batch created: {response.batch_id}")

    # Wait for the batch to complete
    result = await client.wait_for_completion_batch(response.batch_id)
    for item in result.items:
        print(f"Batch item: {item.response.message}")


if __name__ == "__main__":
    try:
        asyncio.run(main())
    except Exception as e:
        print(f"Error: {e}")
        exit(1) 